core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -1040,10 +1041,52 @@ static void loadDirectoryMetadata(RpcContext rpcContext, LockedInodePath inodePa
     createDirectoryContext.setWriteType(WriteType.THROUGH);
     MountTable.Resolution resolution = mountTable.resolve(inodePath.getUri());
 
-    AlluxioURI ufsUri = resolution.getUri();
+    // create the actual metadata
+    loadDirectoryMetadataInternal(rpcContext, createDirectoryContext, context, inodePath,
+        resolution.getUri(), resolution.getUfsClient(), fsMaster, resolution.getShared());
+  }
+
+  /**
+   * Loads metadata for the directory that will be the mount point.
+   *
+   * This difference between this method and
+   * {@link InodeSyncStream#loadDirectoryMetadata(RpcContext, LockedInodePath, LoadMetadataContext,
+   * MountTable, DefaultFileSystemMaster)} is that it accepts more specific parameters(i.e
+   * . isShared, ufsUri and ufsClient) and doesn't require the MountTable as its parameter.
+   */
+  static void loadMountPointDirectoryMetadata(RpcContext rpcContext, LockedInodePath inodePath,
```

Comment:
can you merge this with `loadDirectoryMetadata()` more? Seems only L1075 is the only difference and you can just merge the rest of code.